packages/react/src/transactions/adapters/CollectPublicationCallGateway.ts

```diff
@@ -4,37 +4,70 @@ import {
   CreateCollectTypedDataVariables,
   LensApolloClient,
   omitTypename,
+  ProxyActionData,
+  ProxyActionDocument,
+  ProxyActionRequest,
+  ProxyActionVariables,
 } from '@lens-protocol/api-bindings';
 import { lensHub } from '@lens-protocol/blockchain-bindings';
-import { Nonce } from '@lens-protocol/domain/entities';
-import { CollectRequest } from '@lens-protocol/domain/use-cases/publications';
-import { IOnChainProtocolCallGateway } from '@lens-protocol/domain/use-cases/transactions';
-import { invariant } from '@lens-protocol/shared-kernel';
+import { Nonce, ProxyTransaction } from '@lens-protocol/domain/entities';
+import { CollectRequest, FreeCollectRequest } from '@lens-protocol/domain/use-cases/publications';
+import {
+  BroadcastingError,
+  IOnChainProtocolCallGateway,
+  ISignlessSubsidizedCallRelayer,
+} from '@lens-protocol/domain/use-cases/transactions';
+import {
+  ChainType,
+  ILogger,
+  PromiseResult,
+  assertError,
+  failure,
+  getID,
+  invariant,
+  success,
+} from '@lens-protocol/shared-kernel';
 
 import { UnsignedProtocolCall } from '../../wallet/adapters/ConcreteWallet';
+import { ITransactionFactory } from './ITransactionFactory';
+import { ProxyReceipt } from './ProxyReceipt';
 import { Data, SelfFundedProtocolTransactionRequest } from './SelfFundedProtocolTransactionRequest';
 
-export class CollectPublicationCallGateway implements IOnChainProtocolCallGateway<CollectRequest> {
-  constructor(private apolloClient: LensApolloClient) {}
+export class CollectPublicationCallGateway
```

Comment:
Naming wise I would probably not retain the word "call" as it's mean to signify "protocol call" entity.

Maybe just `CollectPublicationGateway`?